frontend/app/routes/_public+/apply+/$id+/demographics-part1.tsx

```diff
@@ -26,58 +32,165 @@ export const meta: MetaFunction<typeof loader> = mergeMeta((args) => {
+    throw new Response(`Unexpected 'Other' gender type: ${FIRST_NATIONS_YES_TYPE_ID}`, { status: 500 });
+  }
+
+  return json({ id, state: state.demographicsPart1, bornTypes, disabilityTypes, firstNationsTypes, firstNationsEnumTypes, firstNationsYesCode });
 }
 
-export async function action({ request }: ActionFunctionArgs) {
-  return json('');
+export async function action({ request, params }: ActionFunctionArgs) {
+  const applyFlow = getApplyFlow();
+  const { id } = await applyFlow.loadState({ request, params });
+
+  const formSchema = z.object({
+    bornType: z.string().trim().optional(),
+    //firstNationsEnum: z.string().min(1, { message: 'empty-field' }).optional(),
+    firstNations: z.string().trim().optional(),
+    disabilityType: z.string().trim().optional(),
+  });
+
+  const formData = Object.fromEntries(await request.formData());
+  const parsedDataResult = formSchema.safeParse(formData);
+  if (!parsedDataResult.success) {
+    return json({
+      errors: parsedDataResult.error.format(),
+      formData: formData as Partial<z.infer<typeof applyFlow.demographicsPart1StateSchema>>,
+    });
+  }
+
+  const sessionResponseInit = await applyFlow.saveState({
+    request,
+    params,
+    state: { demographicsPart1: parsedDataResult.data },
+  });
+  return redirect(`/apply/${id}/demographics-part2`, sessionResponseInit);
 }
 
 export default function DemographicsPart1() {
-  const { bornTypes, disabilityTypes } = useLoaderData<typeof loader>();
+  const { bornTypes, disabilityTypes, firstNationsTypes, firstNationsEnumTypes, firstNationsYesCode, state, id } = useLoaderData<typeof loader>();
   const { i18n, t } = useTranslation(handle.i18nNamespaces);
+  const [firstNationsYesChecked, setFirstNationsYesChecked] = useState(state?.firstNations === firstNationsYesCode.id);
+  const actionData = useActionData<typeof action>();
+  const errorSummaryId = 'error-summary';
+  function firstNationsHandler() {
+    setFirstNationsYesChecked(true);
+  }
+
+  function nonFirstNationsHandler() {
+    setFirstNationsYesChecked(false);
+  }
+
+  const nonFirstNationsTypeOptions: InputRadiosProps['options'] = firstNationsTypes
+    .filter((firstNationsType) => firstNationsType.id !== firstNationsYesCode.id)
+    .map((firstNationsType) => ({
+      children: getNameByLanguage(i18n.language, firstNationsType),
+      value: firstNationsType.id,
+      defaultChecked: state?.firstNations === firstNationsType.id,
+      onClick: nonFirstNationsHandler,
+    }));
+
+  const options: InputRadiosProps['options'] = [
+    {
+      children: getNameByLanguage(i18n.language, firstNationsYesCode),
+      value: firstNationsYesCode.id,
+      defaultChecked: state?.firstNations === firstNationsYesCode.id,
+      append: firstNationsYesChecked && (
+        <InputCheckboxes
+          id="firstNationsEnum"
+          name="firstNationsEnum"
+          legend={t('apply:demographics-oral-health-questions.part1.first-nations-blank')}
```

Comment:
This blank `legend` seems odd. A few suggestions/options:

1. Specify a `legend` in both languages and use [Tailwind's `sr-only`](https://tailwindcss.com/docs/screen-readers#screen-reader-only-elements) CSS class on the `legend`
2. Ask business to specify a legend text for us to display
3. Use individual `<InputCheckbox />` instead of the `<InputCheckboxes />`, although this might result in accessibility issues if we don't specify a `<fieldset>` and `<legend>`

Reply:
legend has a text for now, need to validate with business clients